packages/matchbox/src/components/ActionList/Action.tsx

```diff
@@ -36,54 +35,105 @@ type LinkProps = {
   is?: 'link';
 } & LinkActionProps &
   React.ComponentPropsWithoutRef<'a'> &
-  BaseProps;
+  ActionListActionProps;
 
 type ButtonProps = {
-  is: 'button';
-} & React.ComponentPropsWithoutRef<'button'> &
-  BaseProps;
-
-type ActionProps = LinkProps | ButtonProps;
-
-const Action = React.forwardRef<HTMLAnchorElement, ActionProps>(function Action(
-  props: ActionProps,
-  userRef,
-) {
-  const { content, children, disabled, helpText, is = 'link', selected, ...action } = props;
-
-  const linkContent = React.useMemo(() => {
-    return (
-      <Box as="span" alignItems="flex-start" display="flex">
-        <Box as="span" flex="1" fontSize="300" lineHeight="300">
-          {content}
-          {children}
-        </Box>
-        {selected && (
-          <Box as="span" color="blue.700">
-            <CheckBox size={20} />
-          </Box>
-        )}
-        {helpText && <HelpText>{helpText}</HelpText>}
+  is?: 'button';
+} & LinkActionProps &
+  React.ComponentPropsWithoutRef<'button'> &
+  ActionListActionProps;
+
+function Content(props: ActionListActionProps): JSX.Element {
+  const { content, children, selected, helpText } = props;
+
+  return (
+    <Box as="span" alignItems="flex-start" display="flex">
+      <Box as="span" flex="1" fontSize="300" lineHeight="300">
+        {content}
+        {children}
       </Box>
-    );
-  }, [content, selected, helpText]);
+      {selected && (
+        <Box as="span" color="blue.700">
+          <CheckBox size={20} />
+        </Box>
+      )}
+      {helpText && <HelpText>{helpText}</HelpText>}
+    </Box>
+  );
+}
+
+// TODO fix this any
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const ButtonAction = React.forwardRef<any, any>(function ButtonAction(props, userRef) {
```

Comment:
Was having trouble with these components. I split them up so the component that is exposed to the end-users is correctly typed, its just these internal ones that aren't